### DISCUSSION TITLE
Allow 4 byte quote delimited literals to be used for Runes

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#7340] [Language Ideas]
The escape sequence `\U0001F01C` cane be used today within a string to represent a surrogate pair. 

```cs
string myString = "In the game of mahjong \U0001F01C denotes the Four of circles";
```

I'm wondering if it would be considered useful to allow this:

```cs
Rune pair = '\U0001F01C';
```
Although the single quote is currently used to represent a char literal, it can also be used to represent a rune like above and there's no ambiguity or backward compatibility problem that I can see. This could avoid the overhead to of needing to use "new" to initialize such a rune.

**Comment by CyrusNajmabadi** [comment]
@stephentoub (and anyone you want to pull in from the runtime in this area.  It does seem like it would be nice to be able to directly create runes from literals (like you can with chars).  We could even consider something like `'\u1234'r` here.  

Curious though how important you think this is to make it convenient to go directly with the combined pair, instead of the individual pieces.

**Reply by CyrusNajmabadi** [reply]
Tagging @tmat as well for interest in the language space here.

**Comment by tannergooding** [comment]
> This could avoid the overhead to of needing to use "new" to initialize such a rune.

It's worth noting there is no real "overhead" here. Particularly for constant inputs the JIT optimizes this down to constant initialization already. Likewise, it is not a primitive and so has no constant representation in IL. Thus, `Rune pair = '\U0001F01C';` would simply be syntax sugar for `Rune pair = new Rune(0x0001F01C)` anyways.

It is really just a convenience factor for constructing the values. That itself may be a good enough reason, just that it's important to understand the types of limits and considerations that may come along with that.

**Reply by CyrusNajmabadi** [reply]
it would also be interesting whether or not we could/would consider this a constant or not.

**Reply by tannergooding** [reply]
I think it would have many of the same limitations as `decimal`. Would be nice if there were some more general purpose way to support “constants” for user defined structs though
